### PR TITLE
Feature: fetch with including specific or all keys

### PR DESCRIPTION
**Pull request by cbaker6**
- [x] Adds `fetch(includeKeys: [String]? = nil)` and ``fetchPublisher(includeKeys: [String]? = nil)`` to all `ParseObjects`. 
- [x] Documentation
- [x] Testcases
- [x] Playgrounds

Non breaking...

### CODE REVIEW

**Comment by cbaker6** [comment]
@dplewis a question for you about your PR https://github.com/parse-community/Parse-SDK-JS/pull/631

Can you tell me how the URI should look? My interpretation is that they should be query parameters, so the URI's would look something like:

```
http://localhost:1337/1/installations/pqHEgVBo4HnreKMiC0Y1CXRCXqb40WUg?include=score
```

and 

```
http://localhost:1337/1/installations/pqHEgVBo4HnreKMiC0Y1CXRCXqb40WUg?include=*
```

I'm sure I'm doing something wrong as I'm still getting back pointers.

**Comment by cbaker6** [comment]
Maybe I misread, was the PR never merged on the server, only in the JS SDK?

**Comment by dplewis** [comment]
FetchwithInclude is basically like a query with include so it should work on the server.

Are you setting the pointer fields in your test case?

**Comment by cbaker6** [comment]
> Are you setting the pointer fields in your test case?

Yes, my output from connecting to an actual server in playgrounds is below:
```
http://localhost:1337/1/installations/pqHEgVBo4HnreKMiC0Y1CXRCXqb40WUg?include=score
Successfully fetched user with score key: _Installation ({"updatedAt":{"__type":"Date","iso":"2021-02-01T19:57:01.877Z"},"customKey":"myCustomInstallationKey2","parseVersion":"0.0.1","deviceType":"osx","installationId":"de66d713-2bd7-4b59-8389-145706e037fd","localeIdentifier":"en-US","appIdentifier":"com.apple.dt.Xcode.PlaygroundStub-macosx","timeZone":"America\/New_York","appName":"com.apple.dt.Xcode.PlaygroundStub-macosx","targetScore":{"__type":"Pointer","className":"GameScore","objectId":"Wzpfpc5sXQyMdDdgb6YJHOAEAPFuvij4"},"objectId":"pqHEgVBo4HnreKMiC0Y1CXRCXqb40WUg","appVersion":"1","createdAt":{"__type":"Date","iso":"2021-02-01T19:57:01.877Z"},"score":{"__type":"Pointer","className":"GameScore","objectId":"DvyVRuPphLnBMCZAGWpU27nhAHkANayj"}})
```

```
http://localhost:1337/1/installations/pqHEgVBo4HnreKMiC0Y1CXRCXqb40WUg?include=*
Successfully fetched user with all keys: _Installation ({"updatedAt":{"__type":"Date","iso":"2021-02-01T19:57:01.877Z"},"customKey":"myCustomInstallationKey2","parseVersion":"0.0.1","deviceType":"osx","installationId":"de66d713-2bd7-4b59-8389-145706e037fd","localeIdentifier":"en-US","appIdentifier":"com.apple.dt.Xcode.PlaygroundStub-macosx","timeZone":"America\/New_York","appName":"com.apple.dt.Xcode.PlaygroundStub-macosx","targetScore":{"__type":"Pointer","className":"GameScore","objectId":"Wzpfpc5sXQyMdDdgb6YJHOAEAPFuvij4"},"objectId":"pqHEgVBo4HnreKMiC0Y1CXRCXqb40WUg","appVersion":"1","createdAt":{"__type":"Date","iso":"2021-02-01T19:57:01.877Z"},"score":{"__type":"Pointer","className":"GameScore","objectId":"DvyVRuPphLnBMCZAGWpU27nhAHkANayj"}})
```

**Comment by cbaker6** [comment]
> FetchwithInclude is basically like a query with include so it should work on the server.

To this point, when fetching ` object, I use .GET and the URI as stated in the REST API docs, but for `fetchAll`, I use query. I can easily get this working if I use a query for fetching a single object, since I know that works. Is that what it's doing in JS?

**Comment by dplewis** [comment]
I've never tried it with the installation class. I can run some tests

**Comment by cbaker6** [comment]
I tried it with `User` first, but got the same results. I haven't tried it with a regular parse object on the server.

It's possible I'm making the URI incorrectly.

**Comment by cbaker6** [comment]
@dplewis looks like I had it working. The problem was my debugDescription for print was encoding ParseObjects as back to pointers.

**Comment by cbaker6** [comment]
@TomWFox can you review the docs? It's basically the same line added for all fetch methods

File: Sources/ParseSwift/Objects/ParseInstallation+combine.swift [hunk 1]
```diff
@@ -18,14 +18,18 @@ public extension ParseInstallation {
     /**
      Fetches the `ParseInstallation` *aynchronously* with the current data from the server
      and sets an error if one occurs. Publishes when complete.
-
+     - parameter includeKeys: The name(s) of the key(s) to include that are
+     `ParseObjects`. Use `["*"]` to include all keys. This is similar to `include` and
+     `includeAll` for `Query`.
```
Comment (TomWFox) [inline Sources/ParseSwift/Objects/ParseInstallation+combine.swift:23]:
```suggestion
     - parameter includeKeys: The name(s) of the key(s) to include that are
     `ParseObject`s. Use `["*"]` to include all keys. This is similar to `include` and
     `includeAll` for `Query`.
```

**Review by TomWFox — CHANGES_REQUESTED** [review]

**Comment by dplewis** [comment]
Nice! I should add this to the iOS SDK

**Pull request merged**